lib/rake/extensiontask.rb

```diff
@@ -79,6 +79,35 @@ def cross_config_options(for_platform=nil)
       end.flatten
     end
 
+    def make_makefile_cmd(root_path, tmp_path, extconf, siteconf_path, cross_platform) # :nodoc:
+      options = @config_options.dup
+
+      # include current directory
+      include_dirs = ['.'].concat(@config_includes).uniq.join(File::PATH_SEPARATOR)
+      cmd = [Gem.ruby, "-I#{include_dirs}", "-r#{File.basename(siteconf_path)}"]
+
+      # build a relative path to extconf script
+      abs_tmp_path = (Pathname.new(root_path) + tmp_path).realpath
+      abs_extconf = (Pathname.new(root_path) + extconf).realpath
+
+      # now add the extconf script
+      cmd << abs_extconf.relative_path_from(abs_tmp_path).to_s
+
+      if cross_platform
+        options.push(*cross_config_options(cross_platform))
+      end
+
+      # add options to command
+      cmd.push(*options)
+
+      # add any extra command line options
+      unless extra_options.empty?
+        cmd.push(*extra_options)
+      end
```

Comment:
How about simplifying this like others?

```suggestion
      cmd.push(*extra_options)
```

Reply:
I added a commit to clean up this method in a different way. I don't feel strongly and can do it as you requested if you prefer.